pkg/volume/csi/csi_plugin.go

```diff
@@ -103,42 +103,41 @@ func RegistrationCallback(pluginName string, endpoint string, versions []string,
 		endpoint = socketPath
 	}
 
-	// Storing endpoint of newly registered CSI driver into the map, where CSI driver name will be the key
-	// all other CSI components will be able to get the actual socket of CSI drivers by its name.
-	csiDrivers.Lock()
-	defer csiDrivers.Unlock()
-	csiDrivers.driversMap[pluginName] = csiDriver{driverName: pluginName, driverEndpoint: endpoint}
+	func() {
+		// Storing endpoint of newly registered CSI driver into the map, where CSI driver name will be the key
+		// all other CSI components will be able to get the actual socket of CSI drivers by its name.
+		csiDrivers.Lock()
+		defer csiDrivers.Unlock()
+		csiDrivers.driversMap[pluginName] = csiDriver{driverName: pluginName, driverEndpoint: endpoint}
```

Comment:
Previously the lock was being held across the entire function instead of just the map update.  Could there be any issues if this map has an entry but we haven't finished registering the plugin yet?  Should we update the map at the very end?

Reply:
In both the previous and the current version, the lock is only held on write, all within this `RegistrationCallback()` function. The map has to be updated before the `csi.NodeGetInfo()` call unfortunately because it relies on the driver info to call out to gRPC.

However, when plugin unregistration is implemented in the future, depending on how it's implemented, there could be a race between calls of `RegistrationCallback()`. Will update to lock the entire function once.

Reply:
Discussed with @msau42 offline. For now we'll keep the map locked as short as possible. When unregistration is implemented we'll reconsider again.

Will add comments explaining why it's OK to keep locking short.